stdlib/public/RemoteInspection/TypeRefBuilder.cpp

```diff
@@ -93,6 +93,7 @@ RemoteRef<char> TypeRefBuilder::readTypeRef(uint64_t remoteAddr) {
 /// Load and normalize a mangled name so it can be matched with string equality.
 llvm::Optional<std::string>
 TypeRefBuilder::normalizeReflectionName(RemoteRef<char> reflectionName) {
+  auto checkpoint = pushNodeFactoryCheckpoint();
```

Comment:
A lot of code that does this seems to follow the pattern
```
auto checkpoint = pushNodeFactoryCheckpoint();
...
popNodeFactoryCheckpoint(checkpoint);
```
which makes me wonder whether we should have a
```
template <typename T, typename F>
T withScopedNodes(F func) {
  auto checkpoint = pushNodeFactoryCheckpoint();
  T result = func();
  popNodeFactoryCheckpoint(checkpoint);
  return result;
}
```
or similar (obviously that's written OTOH in a Github comment, but you get the idea).

Reply:
RAII would be more idiomatic in C++. I've gone ahead and added a little RAII wrapper and switched over to that. The scoping makes things less convenient in a couple of places where we declare locals that get used later, but overall it's nicer.

Reply:
:-) I seems to have Swift on the brain at the moment. Yes, RAII is more idiomatic in C++.